### PR TITLE
Upi_Info_Card: Created a Upi Info card component

### PR DESCRIPTION
**Pull request by NileshSingh9245**
### Release

1.0.0
https://blackminystudio.atlassian.net/browse/TPA-47

### Description

- Created a new component Upi info card,

### Screenshots

image

### Tips

Use this code in main.dart for testing.
```Dart
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) => const Scaffold(
  backgroundColor: theme.colors.light,
        body: Center(
          child: Padding(
            padding: EdgeInsets.all(4.0),
            child: UpiInfoCard(), 
          ),
        ),
      );
}

```

---

### IMPORTANT:

- (REQUIRED) Coverage Screenshots for the tests written
- (REQUIRED) UI Screenshots for the UI implementation written

### Author preparation checklist

- [x] Current branch is rebased with latest 'main' branch
- [x] PR is targeting only one issue
- [x] Commit messages are descriptive, consistent, and follow the project's guidelines.
- [ ] Branch name matches with at least `<jira link reference>-<short change description>`
- [x] Primary reviewer is set as assignee to the PR
- [x] No unnecessary changes on `pubspec.lock`
- [x] All added TODOs have associated Github issue references
- [x] \*.iml and, pubsepc_overrides.yaml are not checkedin

### Primary reviewer checklist

- [x] All items in the author checklist is checked or not applicable
- [x] All files including .lock and assets are reviewed
- [x] Reasonable UI flows/cases related to the change are manually tested
- [x] No commented-out code or unnecessary debug statements.
- [x] No spelling or grammatical errors in code comments or documentation.
- [x] No sensitive data (e.g., API keys, credentials) is hardcoded or exposed.
- [x] The branch is up-to-date with the target branch (e.g., staging).

### Pre-merge checklist

- [ ] Two approvals received. (Changes affecting automatic PR verification, like new linter rules, formatting rules, Github workflow configurations)

---

### CODE REVIEW

File: lib/widgets/upi_info_card.dart [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    ),
+                  ),
+                  SizedBox(
+                    width: theme.sizing.width.s32,
+                    child: TextField(
+                      controller: _amountController,
+                      keyboardType: TextInputType.number,
+                      onChanged: (value) => setState(() {}),
+                      decoration: InputDecoration(
+                        hintText: 'Enter Amount',
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart:61]: All the hard-coded texts need to be written in string_constants.dart inside widgets folder

File: lib/widgets/upi_info_card.dart [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                ],
+              ),
+              SizedBox(height: theme.sizing.width.s2),
+              const Divider(),
+              SizedBox(height: theme.sizing.width.s2),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  Text(
+                    'Message:',
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart:80]: write in string_constants.dart and use it

File: lib/widgets/upi_info_card.dart [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                    style: theme.textStyle.bodyBold.copyWith(
+                      color: theme.colors.contrastMedium,
+                    ),
+                  ),
+                  Flexible(
+                    child: TextField(
+                      controller: _messageController,
+                      onChanged: (value) => setState(() {}),
+                      decoration: InputDecoration(
+                        hintText: 'Enter Message',
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart:90]: write in string_constants.dart and use it

File: lib/widgets/upi_info_card.dart [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                            .copyWith(color: theme.colors.contrastLow),
+                      ),
+                      style: theme.textStyle.titleBold
+                          .copyWith(color: theme.colors.contrastDark),
+                      textAlign: TextAlign.right,
+                    ),
+                  ),
+                ],
+              ),
+              SizedBox(height: theme.sizing.width.s2),
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart:73]: theme.spacing.width.s12 as per figma

File: lib/widgets/upi_info_card.dart [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+              const Divider(),
+              SizedBox(height: theme.sizing.width.s2),
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart:75]: theme.spacing.width.s12 as per figma

**Review by quiteachaos — DISMISSED** [review]

File: lib/widgets/upi_info_card.dart (hunk 1 above)
Comment (NileshRajput1) [reply lib/widgets/upi_info_card.dart]: Done

File: lib/widgets/upi_info_card.dart (hunk 2 above)
Comment (NileshRajput1) [reply lib/widgets/upi_info_card.dart]: Done

File: lib/widgets/upi_info_card.dart (hunk 3 above)
Comment (NileshRajput1) [reply lib/widgets/upi_info_card.dart]: Done

File: lib/widgets/upi_info_card.dart (hunk 5 above)
Comment (NileshRajput1) [reply lib/widgets/upi_info_card.dart]: done

File: lib/widgets/upi_info_card.dart [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    _amountController.dispose();
+    _messageController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Card(
```
[the remark this answers is not in the document]
Comment (NileshRajput1) [reply lib/widgets/upi_info_card.dart]: done

File: lib/widgets/upi_info_card.dart [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+      color: theme.colors.contrastLight,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.circular(theme.borderradius.medium),
+      ),
+      child: Container(
```
[the remark this answers is not in the document]
Comment (NileshSingh9245) [reply lib/widgets/upi_info_card.dart:84]: Done

**Review by quiteachaos — DISMISSED** [review]

**Review by satyabratanayak — DISMISSED** [review]

**Comment by NileshSingh9245** [comment]
**Use this code in main.dart for testing .**

```
class HomeScreen extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final theme = Theme.of(context);
    return Scaffold(
      backgroundColor: theme.colors.light,
      body: const Center(
          child: Padding(
        padding: EdgeInsets.symmetric(horizontal: 40),
        child: UpiInfoCard(),
      )),
    );
  }
}
```


**below is the list of addressed changes:**

> - Fixed card height and alignment to match Figma design.
> - Ensured '₹' symbol is visible in the Amount field.
> - Restricted Amount input to two decimal places and digits only using InputFormatters.
> - Set consistent widths for both text fields; added ellipsis for overflow text.
> - Implemented FocusNode to shift focus to the Message field after pressing 'Enter' in Amount field.
> - Updated Divider thickness to 1px and color to contrastLow using a Container.

**Review by quiteachaos — DISMISSED** [review]

**Review by satyabratanayak — DISMISSED** [review]

File: lib/widgets/upi_info_card.dart [hunk 8]
```diff
@@ -0,0 +1,252 @@
...
+
+  String _ensureRupeeSymbol(String text) {
+    if (text.isNotEmpty && !text.startsWith(ruppeeSymbol)) {
+      text = '$ruppeeSymbol$text';
+    }
+    return text;
+  }
+}
+
+// TODO:
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart]: Remove the commented code

File: lib/widgets/string_constants.dart [hunk 9]
```diff
@@ -1,7 +1,19 @@
+// TacklePay App
 const appName = 'Tackle Pay';
+const ruppeeSymbol = '₹';
+
 const incomeText = 'Income';
 const expenseText = 'Expense';
 const checkZero = '0';
 const checkMinus = '-';
+const newExpense = 'New Expense';
```
Comment (quiteachaos) [inline lib/widgets/string_constants.dart:9]: newExpenseText*

File: lib/widgets/string_constants.dart [hunk 10]
```diff
@@ -1,7 +1,19 @@
+// TacklePay App
 const appName = 'Tackle Pay';
+const ruppeeSymbol = '₹';
```
Comment (quiteachaos) [inline lib/widgets/string_constants.dart]: rupeeSymbol

File: lib/widgets/upi_info_card.dart [hunk 11]
```diff
@@ -0,0 +1,252 @@
...
+    }
+    return _formatIndianNumber(text);
+  }
+
+  String _formatIndianNumber(String number) {
+    if (number.isEmpty) return number;
+    final length = number.length;
+    if (length <= 3) return number;
+
+    // Take the last three digits
```
Comment (quiteachaos) [inline lib/widgets/upi_info_card.dart]: redundant comment should be removed

**Comment by NileshSingh9245** [comment]
everything looks good to me

**Review by quiteachaos — DISMISSED** [review]

**Review by satyabratanayak — DISMISSED** [review]

File: lib/widgets/upi_info_card.dart (hunk 11 above)
Comment (satyabratanayak) [reply lib/widgets/upi_info_card.dart]: done

File: lib/widgets/string_constants.dart (hunk 10 above)
Comment (satyabratanayak) [reply lib/widgets/string_constants.dart]: done

File: lib/widgets/upi_info_card.dart (hunk 8 above)
Comment (satyabratanayak) [reply lib/widgets/upi_info_card.dart]: done

**Review by quiteachaos — APPROVED** [review]

**Review by satyabratanayak — APPROVED** [review]

**Pull request merged**